rustls/src/client/hs.rs

```diff
@@ -814,16 +813,22 @@ impl<C: CryptoProvider> ExpectServerHelloOrHelloRetryRequest<C> {
 
         let key_share = match req_group {
             Some(group) if group != offered_key_share.group() => {
-                match KeyExchange::start(group, &config.kx_groups) {
-                    Ok(kx) => kx,
-                    Err(KeyExchangeError::UnsupportedGroup) => {
+                let skxg = match config
```

Comment:
Maybe turn this into a method on `config` that both this and `ExpectServerDone` can use?